mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java

```diff
@@ -127,10 +131,17 @@ public WidgetAdapter(Context context, Connection connection,
         TypedValue tv = new TypedValue();
         context.getTheme().resolveAttribute(R.attr.chartTheme, tv, true);
         mChartTheme = tv.string;
+
+        mRecyclerView = recyclerView;
+        mEmptyPageView = emptyPageView;
     }
 
     public void update(List<Widget> widgets, boolean forceFullUpdate) {
         boolean compatibleUpdate = true;
+        boolean emptyPage = widgets.size() == 0;
+
+        mRecyclerView.setVisibility(emptyPage ? View.GONE : View.VISIBLE);
+        mEmptyPageView.setVisibility(emptyPage ? View.VISIBLE : View.GONE);
```

Comment:
Should do this in the fragment, otherwise you have an inversion of control flow (fragment owns and controls  adapter, not the other way around). `update()` is called from fragment anyway.